src/System.CommandLine/Invocation/ParseErrorAction.cs

```diff
@@ -59,7 +60,30 @@ private static void WriteErrorDetails(ParseResult parseResult)
 
     private static void WriteHelp(ParseResult parseResult)
     {
-        new HelpAction().Invoke(parseResult);
+        var availableHelpOptions =
```

Comment:
I have fixed this bug once: https://github.com/dotnet/command-line-api/pull/2211, how is it possible that it's back? Some merge issue?

Since `ParseResult` exposes `RootCommandResult` and `RootCommand` should be the only command having it, we can simplify it to:

```cs
HelpOption helpOption = parseResult.RootCommandResult.Command.Options.FirstOrDefault(option => option is HelpOption) as HelpOption ?? new HelpOption();
```

Reply:
> Since ParseResult exposes RootCommandResult and RootCommand should be the only command having it...

This isn't the case. Different help options can be defined at different levels of the hierarchy.